### PR TITLE
eventql 0.3.2 (new formula)

### PR DESCRIPTION
**Pull request by bartleusink**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/eventql.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Eventql < Formula
+  desc "Database for large-scale event analytics"
+  homepage "https://eventql.io"
+  url "http://dl.eventql.io/eventql/v0.3.2/eventql-0.3.2.tgz"
```
Comment (zmwangx) [inline Formula/eventql.rb]: Let's try to use TLS: https://github.com/eventql/eventql/releases/download/v0.3.2/eventql-0.3.2.tgz.

File: Formula/eventql.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  head "https://github.com/eventql/eventql.git"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+
+  def install
+    ENV.deparallelize
+    system "./autogen.sh" if build.head?
+
```
Comment (zmwangx) [inline Formula/eventql.rb]: Remove the unnecessary blank line. Same below.

File: Formula/eventql.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    system "./configure", "--disable-dependency-tracking",
+                          "--prefix=#{prefix}"
+
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/evql", "--help"
```
Comment (zmwangx) [inline Formula/eventql.rb]:
```
system bin/"evql", "--help"
```

But is it possible to write a more substantial test that actually tries to do something with `evql`?

**Review by zmwangx — COMMENTED** [review]
A few comments otherwise 👍

File: Formula/eventql.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    ENV.deparallelize
+    system "./autogen.sh" if build.head?
+    system "./configure", "--disable-dependency-tracking",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    begin
+      system "evqld", "--standalone", "--datadir", testpath, "--daemonize", "--pidfile", "pidfile"
```
Comment (zmwangx) [inline Formula/eventql.rb]: Line too long, split into two.

File: Formula/eventql.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+
+  def install
+    ENV.deparallelize
+    system "./autogen.sh" if build.head?
+    system "./configure", "--disable-dependency-tracking",
+                          "--prefix=#{prefix}"
```
Comment (zmwangx) [inline Formula/eventql.rb]: Don't need a line break here, we only have 79 characters.

File: Formula/eventql.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
+class Eventql < Formula
+  desc "Database for large-scale event analytics"
+  homepage "https://eventql.io"
+  url "https://github.com/eventql/eventql/releases/download/v0.3.2/eventql-0.3.2.tgz"
+  sha256 "d235f3e78fa5f6569fc2db94161d3e3f9cb71dc0646e341acd91814cefd23640"
+  head "https://github.com/eventql/eventql.git"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
```
Comment (zmwangx) [inline Formula/eventql.rb]: I suppose these should be `=> :build if build.head?`.

**Review by zmwangx — CHANGES_REQUESTED** [review]
Thanks for the fixes and the new test. A few more comments.

**Comment by bartleusink** [comment]
Thanks for the feedback Zhiming!

File: Formula/eventql.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  url "https://github.com/eventql/eventql/releases/download/v0.3.2/eventql-0.3.2.tgz"
+  sha256 "d235f3e78fa5f6569fc2db94161d3e3f9cb71dc0646e341acd91814cefd23640"
+  head "https://github.com/eventql/eventql.git"
+
+  depends_on "autoconf" => :build if build.head?
+  depends_on "automake" => :build if build.head?
+  depends_on "libtool" => :build if build.head?
+
+  def install
+    ENV.deparallelize
```
Comment (zmwangx) [inline Formula/eventql.rb]: Is this necessary? Did you report the parallel make problem upstream?

**Review by zmwangx — COMMENTED** [review]
One last question (hopefully) which didn't come through last time for whatever reason.

**Review by zmwangx — APPROVED** [review]

File: Formula/eventql.rb (hunk 7 above)
Comment (bartleusink) [inline Formula/eventql.rb]: Without `ENV.deparallelize` I ran into this issue: https://github.com/eventql/eventql/issues/180

Comment (zmwangx) [inline Formula/eventql.rb]: Sure. Could you please add a comment above this line linking to the issue?

**Comment by zmwangx** [comment]
FYI the test hangs locally on my machine (specifically the `evqld` command hangs, although running the command outside of `brew test` was fine). It doesn't appear to be a sandbox issue either. Were you able to run the test successfully?

**Comment by zmwangx** [comment]
Indeed, test expired on the CI too.

**Comment by bartleusink** [comment]
The test runs fine on my machine, but I remember the first time it triggered an OSX dialogue asking me whether I wanted `evqld` to open a port.
Could this be related to firewall settings?

**Comment by bartleusink** [comment]
I've changed the way `evqld` is started, which seems to fix the hanging issue.

**Comment by zmwangx** [comment]
Cool, I'm happy. Let's wait for a while to see if other maintainers have concerns.

File: Formula/eventql.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    system "./autogen.sh" if build.head?
+    system "./configure", "--disable-dependency-tracking", "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    begin
+      pid = fork do
+        exec "#{bin}/evqld", "--standalone", "--datadir", testpath
```
Comment (zmwangx) [inline Formula/eventql.rb]: `bin/"evqld"`.

File: Formula/eventql.rb [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+      end
+      sleep 1
+      system "#{bin}/evql", "--database", "test", "-e", "SELECT 42;"
```
Comment (zmwangx) [inline Formula/eventql.rb]: Same here.

**Review by zmwangx — CHANGES_REQUESTED** [review]
Sorry, this is the last, I promise.

**Review by zmwangx — APPROVED** [review]

**Comment by zmwangx** [comment]
Merged as 571583c. Thank you for your contribution to Homebrew!

P.S. Eventually I made another style change: grouping the head deps into a head do block. That one was my fault so I fixed it.

**Comment by bartleusink** [comment]
Thanks for all your help getting this formula merged!

**Pull request closed without merging**